spec/acceptance/apt_spec.rb

```diff
@@ -12,21 +12,27 @@
     it 'should work with no errors' do
       pp = <<-EOS
       class { 'apt':
-        always_apt_update    => true,
-        purge_sources_list   => true,
-        purge_sources_list_d => true,
-        purge_preferences    => true,
-        purge_preferences_d  => true,
-        update_timeout       => '400',
-        update_tries         => '3',
-        sources              => {
+        update => {
+          'frequency' => 'always',
+          'timeout'   => '400',
+          'tries'     => '3',
+        },
+        purge => {
+          'sources.list'   => true,
+          'sources.list.d' => true,
+          'preferences'    => true,
+          'preferences.d'  => true,
+        },
+        sources => {
           'puppetlabs' => {
-            'ensure'     => present,
-            'location'   => 'http://apt.puppetlabs.com',
-            'repos'      => 'main',
-            'key'        => '4BD6EC30',
-            'key_server' => 'pgp.mit.edu',
-          }
+            'ensure'   => present,
+            'location' => 'http://apt.puppetlabs.com',
+            'repos'    => 'main',
+            'key'      => {
+              'id'     => '4BD6EC30',
+              'server' => 'pgp.mit.edu',
+            },
```

Comment:
It's ~~turtles~~ hashes all the way down!

![turtles all the way down](http://upload.wikimedia.org/wikipedia/commons/4/47/River_terrapin.jpg)